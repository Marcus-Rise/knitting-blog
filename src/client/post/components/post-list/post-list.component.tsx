import React from "react";
import { PostListItem } from "../post-list-item";
import styles from "./post-list.module.scss";
import type { IPost } from "../../../../common/post";

interface IProps {
  items: Array<IPost>;
}

const PostList: React.FC<IProps> = (props) => {
  const posts = props.items.map((i, index) => {
    return (
      <div className="col-12" key={index}>
        <PostListItem className={styles.item} {...i} />
      </div>
    );
  });

  return <div className="row">{posts}</div>;
};

export { PostList };
