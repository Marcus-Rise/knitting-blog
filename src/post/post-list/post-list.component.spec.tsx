import { render } from "@testing-library/react";
import { PostList } from "./post-list.component";
import React from "react";
import type { IPost } from "../post.interface";

describe("PostList", () => {
  test("render", () => {
    const item: IPost = {
      title: "Как вязать красиво",
      slug: "slugslugslugslug",
      date: "Ноябрь 11, 2020",
      description:
        "Хобби — неотъемлемая часть нашей жизни. Они помогают привести нервы в порядок, успокоиться и  после трудового дня. Кто-то играет на гитаре, другие читают книги или пишут картины.  Кто-то выбирает вязание — один из самых популярных видов досуга в мире.",
      imageLabel: "Милый кролик",
      imageSrc: "http://test.com",
    };
    const { asFragment } = render(<PostList items={[item]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
