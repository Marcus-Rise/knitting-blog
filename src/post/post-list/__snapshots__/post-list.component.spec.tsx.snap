// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList render 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="item root"
      >
        <h2
          class="title"
        >
          Как вязать красиво
        </h2>
        <p
          class="meta"
        >
          ноябрь 11, 2020
        </p>
        <div
          class="image"
        >
          <div
            class="preview"
          >
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 100%;"
              />
              <img
                alt="Милый кролик"
                decoding="async"
                src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
                srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=320&q=75 320w, /_next/image?url=http%3A%2F%2Ftest.com&w=420&q=75 420w, /_next/image?url=http%3A%2F%2Ftest.com&w=768&q=75 768w, /_next/image?url=http%3A%2F%2Ftest.com&w=1024&q=75 1024w, /_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1200w"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </div>
        <p
          class="imageLabel"
        >
          Милый кролик
        </p>
        <p
          class="description"
        >
          Хобби — неотъемлемая часть нашей жизни. Они помогают привести нервы в порядок, успокоиться и  после трудового дня. Кто-то играет на гитаре, другие читают книги или пишут картины.  Кто-то выбирает вязание — один из самых популярных видов досуга в мире.
        </p>
        <div
          class="row justify-content-center"
        >
          <div
            class="col-auto"
          >
            <div
              class="link"
            >
              Читать далее
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
